fast_depends/core/build.py

```diff
@@ -55,7 +55,7 @@ def build_call_model(
 
     is_call_async = is_coroutine_callable(call)
     if is_sync is None:
-        is_sync = not is_call_async
+        is_sync = not is_call_async and not inspect.isasyncgenfunction(call)
```

Comment:
Can you use our is_async_gen method? It is already using few lines below https://github.com/Lancetnik/FastDepends/blob/bd13af501c8a01f6f11c6faa94e5c0303e7f2bff/fast_depends/core/build.py#L67